lifecycle-operator/webhooks/pod_mutator/pod_mutating_webhook.go

```diff
@@ -93,16 +94,27 @@ func (a *PodMutatingWebhook) Handle(ctx context.Context, req admission.Request)
 	logger.Info(fmt.Sprintf("Pod annotations: %v", pod.Annotations))
 
 	podIsAnnotated := a.isPodAnnotated(pod)
-	logger.Info("Checked if pod is annotated.")
-
 	if !podIsAnnotated {
 		logger.Info("Pod is not annotated, check for parent annotations...")
 		podIsAnnotated = a.copyAnnotationsIfParentAnnotated(ctx, &req, pod)
 	}
 
 	if podIsAnnotated {
-		logger.Info("Resource is annotated with Keptn annotations, using Keptn scheduler")
-		pod.Spec.SchedulerName = "keptn-scheduler"
+		_, gateRemoved := getLabelOrAnnotation(&pod.ObjectMeta, apicommon.SchedulingGateRemoved, "")
+		logger.Info("Resource is annotated with Keptn annotations")
+		if a.SchedulingGatesEnabled {
+			logger.Info("SchedulingGates enabled")
+			if !gateRemoved {
```

Comment:
if the gate has been removed, do we actually need to proceed with checking for the workloads/app, or could we return at this point aleready?

Reply:
good question, I will check that. I didn't want to change the current behavior of the webhook, but it's definitely something that we should look into

Reply:
actually there should be no need for that, if the deployment/pod was changed, the pod is recreated and therefore the whole process starts from the beginning (adding gate, afterwards removing), so the scenario when the gate is removed here and the pod goes through this part of the code is unrealistic, we can then return here

Reply:
got it, thank you for checking this